src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -49,17 +49,23 @@ internal sealed class VariableState
         /// <summary>
         /// Represents the result of visiting an expression.
         /// Contains a result type which tells us whether the expression may be null,
-        /// and an l-value type which tells us whether we can assign null to the expression.
+        /// and an optional slot for tracked variables.
         /// </summary>
+        [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
         private readonly struct VisitResult
         {
-            public readonly TypeSymbolWithAnnotations RValueType;
-            public readonly TypeSymbolWithAnnotations LValueType;
+            public readonly TypeSymbolWithAnnotations Type;
```

Comment:
This approach fundamentally won't work for the feature as specified.  See my most recently submitted PR for one reason (the lvalue is a type with annotations, while the rvalue is a type with a flow state).  The other reason is that what you have as a single int for the slot needs to be a set of ints and a boolean saying whether it/they are exclusive, to handle code such as `(c ? ref x : ref y) = null;` which should set both variables to the nullable state, and also `(c ? ref x : ref y) = "";` which should not affect the state of either variable.

Reply:
@gafter I think this is an incremental step, not the end-state. 
The question of tracking multiple slots in particular will be addressed when we get to the `ref` scenario (ie. it is intentionally out-of-scope for the present PR).
As for reconciling this PR with yours (which clarifies the annotation vs. state distinction), I'll let you figure out with Chuck what makes more sense for sequencing. My take is that we should probably focus on your PR first (bigger change), we can make smaller adjustments on top of it afterwards.